packages/block-library/src/code/index.js

```diff
@@ -68,6 +69,24 @@ export const settings = {
 	edit,
 
 	save( { attributes } ) {
-		return <pre><code>{ attributes.content }</code></pre>;
+		const content = compose(
+			escapeOpeningSquareBrackets,
+			escapeProtocolInIsolatedUrls
+		)( escapeAmpersands( attributes.content || '' ) );
+		return <pre><code>{ content }</code></pre>;
 	},
 };
+
+function escapeAmpersands( content ) {
+	return content.replace( /&/g, '&amp;' );
+}
+
+function escapeOpeningSquareBrackets( content ) {
+	// This replicates the escaping of HTML tags, where
+	// a tag like <strong> becomes &lt;strong>
+	return content.replace( /\[/g, '&#91;' );
+}
+
+function escapeProtocolInIsolatedUrls( content ) {
+	return content.replace( /(^[^.]^https?:)\/\/([^\s]+$[^.]$)/m, '$1&#47;&#47;$2' );
```

Comment:
While the function names are to an extent self-documenting, it still took me more than a few moments to determine what's happening here and why. I think it could be improved with a JSDoc block for the functions and perhaps some unit tests to describe the expected behavior.

I'm still a bit lost on the regular expression itself, specifically the `[^.]` bits, since I would think they'd be expecting some non-period (`.`) content on the lines prior to or following the URL?

Reply:
I modified the RegExp in 4f97f2c so that it matches the RegExp used in PHP. I also documented the functions in d11537f.